kubetest2-gke/deployer/commandutils.go

```diff
@@ -55,7 +55,13 @@ func (d *deployer) prepareGcpIfNeeded(projectID string) error {
 	}
 
 	if err := runWithOutput(exec.Command("gcloud", "config", "set", "project", projectID)); err != nil {
-		return fmt.Errorf("Failed to set project %s : err %v", projectID, err)
+		return fmt.Errorf("failed to set project %s : err %v", projectID, err)
+	}
+
+	// TODO(chizhg):Remove this once we add --enable-workload-identity as the deployer flag,
+	//              then the PROJECT won't need to be specified in the cluster creation command.
```

Comment:
hmm i would prefer to avoid porting over existing kubetest behavior.
Can we directly add the workload identity option?

Reply:
The main reason I'm doing this way is adding the workload identity option will involve changes in 3 different places 😢 
But you are right, that would be way cleaner and this request is not that urgent, so I will add the option in this PR.

Reply:
Added the workload identity option.